tmt/steps/report/reportportal.py

```diff
@@ -38,113 +25,179 @@ class ReportReportPortalData(tmt.steps.report.ReportStepData):
+        headers = {"Authorization": "bearer " + token,
+                   "accept": "*/*",
+                   "Content-Type": "application/json"}
+
+        TMT_env_vars = self.get("env-var") or []
+        attributes = [{'key': k, 'value': v[0]} for k, v in self.step.plan._fmf_context.items()]
+
+        # Communication with RP instance
         with tmt.utils.retry_session() as session:
-            url = f"{server}/api/v1/{project}/launch/import"
-            self.debug(f"Send the report to '{url}'.")
-            response = session.post(
-                url,
-                headers={
-                    "Authorization": "bearer " + token,
-                    "accept": "*/*",
-                    },
-                files={
-                    # The zip filename is used as the launch name in ReportPortal
-                    "file": (launch_name + ".zip", bytestream, "application/zip"),
-                    },
-                )
-
-        # Handle the response
-        try:
-            message = tmt.utils.yaml_to_dict(response.text).get("message")
-        except (tmt.utils.GeneralError, KeyError):
-            message = response.text
-        if not response.ok:
-            raise tmt.utils.ReportError(
-                f"Received non-ok status code from ReportPortal, response text is: {message}")
-        else:
-            self.debug(f"Response code from the server: {response.status_code}")
-            self.debug(f"Message from the server: {message}")
-            self.info("report", "Successfully uploaded.", "yellow")
+
+            # Create a launch
+            self.info("launch", launch_name, color="green")
+            response = session.post(url=f"{url}/launch",
+                                    headers=headers,
+                                    json={"name": launch_name,
+                                          "description": self.step.plan.summary,
+                                          "attributes": attributes,
+                                          "startTime": timestamp()})
+            self.handle_response(response)
+            launch_uuid = tmt.utils.yaml_to_dict(response.text).get("id")
+            assert launch_uuid is not None
+            self.info("    uuid", launch_uuid, "yellow")
+
+            # For each test
+            for result in self.step.plan.execute.results():
+                test = [test for test in self.step.plan.discover.tests()
+                        if test.serialnumber == result.serialnumber][0]
+                if test.contact:
+                    attributes.append({"key": "contact", "value": test.contact[0]})
+                env_vars = [{'key': k, 'value': v} for k, v in test.environment.items()
+                            if not k.startswith("TMT_") or k in TMT_env_vars]
+
+                #  Create a test item
+                self.info("test", result.name, color="green")
+                response = session.post(url=f"{url}/item",
```

Comment:
@KwisatzHaderach  I guess, you suggest differentiating the log levels: fatal, error, warn, info, debug, trace?
Is that correct?
Not sure of any other way to report failure message in RP. But if you provide the source of the example, I can check it.
Anyways I can try to use the log levels to report error messages, if there is any need for them.
Do you have any suggestions what kinds of errors or failures  should be reported in RP this way?

Reply:
Now that I look at the RP REST API I don't see anything that would stand out as what I was looking for, with the xml it used to work by having <failure message='...'/> which had the message that was pulled from logs with the function mentioned above in each test_case item
maybe it's hidden under some field in parameters?

I don't really have access to any RP instance that I could test this on and you surely have more experience with the API so it would be great if you tested if it actually is hidden under some parameter or someplace, otherwise this would be losing functionality over the current implementation. You can test this by running couple tests, some of which fail, with the junit report plugin (or old implementation of this plugin) which produces xml which should be easily importable into RP and then you can get those field names and find out how to get the failures in there.

If this would be too much work and if we want this patch sooner rather than later feel free to create a new issue from this, but the functionality is kind of important for investigation.

Reply:
I think it refers to the log levels I mentioned above. This is how they look, to make sure we talk about the same thing:
![image](https://github.com/teemtee/tmt/assets/80918188/3dbb693c-9f6a-405c-9b8d-906f7fb05e42)
If that's correct, thank you for your observation, I'll try to look into that and let you know.

Reply:
I tried to follow the pattern used in the junit report plugin, so it could look analogous:
![image](https://github.com/teemtee/tmt/assets/80918188/2a36f39f-b3a3-4cc3-b7ab-8ccb6d464110)
Although I didn't really study the structure of the xml, I estimate this was not really built for RP features and this could be a interesting topic for discussion about mapping tmt features with RP log levels.
Originally, I just listed all logs there (output.txt and journal.txt),
following the current xml format I can filter out the failures into 'Error' log level and keep the main log (output.txt) in 'Info' log level. And append the originally liste logs in Trace?
![image](https://github.com/teemtee/tmt/assets/80918188/960206e0-247e-42a9-99e6-4daf60bc3897)
Do you have any suggestions for this, so it is ready for now and it does not lose any functionality over the current implementation?

Reply:
@KwisatzHaderach I tried to follow the pattern used in the junit report plugin, so it could look analogous:
![image](https://github.com/teemtee/tmt/assets/80918188/2a36f39f-b3a3-4cc3-b7ab-8ccb6d464110)
Although I didn't really study the structure of the xml, I estimate this was not really built for RP features and this could be a interesting topic for discussion about mapping tmt features with RP log levels.
Originally, I just listed all logs there (output.txt and journal.txt),
following the current xml format I can filter out the failures into 'Error' log level and keep the main log (output.txt) in 'Info' log level. And append the originally liste logs in Trace?
![image](https://github.com/teemtee/tmt/assets/80918188/960206e0-247e-42a9-99e6-4daf60bc3897)
Do you have any suggestions for this, so it is ready for now and it does not lose any functionality over the current implementation?

Reply:
Nice, thanks for the investigation, IMO it would be best to have just the errors on error level and the whole log with info level as that is apparently what the xml import did.

Reply:
Should I omit all the others log or add them in the last level to have them included? (journal.txt in this case)

Reply:
I guess it is quite the opposite..
Comparing the logs, output.txt is the one printed currently by xml plugin, and it lacks the console information that are in the journal.txt

Reply:
I think @4N0body5 is correct
Currently in the one of our pipeline we move the following error parts 
to testcase message area

Please refer this example
`
<testcase name="auto-osbuild-qemu-rhel9-qa-ostree-aarch64-2869629.3505d8fa:aarch64:/plans/tests" classname="tests">
<failure type="FAIL" message="Test "auto-osbuild-qemu-rhel9-qa-ostree-aarch64-2869629.3505d8fa:aarch64:/plans/tests" failed. Found a bug? Please open an issue in the issue tracker: http://projects.engineering.redhat.com/browse/TFT/issues "/>
<system-out>"The systemout failure"
`
We convert to 
`
<testcase name="auto-osbuild-qemu-rhel9-qa-ostree-aarch64-2869629.3505d8fa:aarch64:/plans/tests" classname="tests"><failure type="FAIL" message="The systemout failure"
`

Otherwise classification tool could not see the failures of info log

Reply:
As you can see in test name
There is an important issue of tests names
for test trending in ReportPortal test name must be identical all the time
`
What we do with external script is changing this
<testcase name="auto-osbuild-qemu-rhel9-qa-ostree-aarch64-2869629.3505d8fa:aarch64:/plans/tests"
`
to 
`
<testcase name="/plans/tests"
`